### PR TITLE
engine: More robust refresh loop.

### PR DESCRIPTION
**Pull request by aluzzardi**
- Random heartbeat (between 30 and 60 seconds).
- Requires 3 failures before marking a node as dead.

Signed-off-by: Andrea Luzzardi <aluzzardi@gmail.com>

### CODE REVIEW

File: cluster/engine.go [hunk 1]
```diff
@@ -20,7 +21,9 @@ import (
 
 const (
 	// Force-refresh the state of the engine this often.
-	stateRefreshPeriod = 30 * time.Second
+	stateRefreshMinRange = 30
```
Comment (aluzzardi) [inline cluster/engine.go]: Not sure about those values

**Comment by vieux** [comment]
LGTM
ping @abronan

File: cluster/engine.go (hunk 1 above)
Comment (abronan) [inline cluster/engine.go]: We can tweak those values as needed. (And ideally be able to change those values when launching the Manager.)

**Comment by abronan** [comment]
LGTM

**Pull request merged**